xbmc/threads/Thread.cpp

```diff
@@ -177,20 +181,48 @@ void CThread::Process()
 
 bool CThread::IsCurrentThread() const
 {
-  return IsCurrentThread(ThreadId());
+  if (m_thread != nullptr)
+    return std::this_thread::get_id() == m_thread->get_id();
+  else
+    return false;
 }
 
 CThread* CThread::GetCurrentThread()
 {
   return currentThread;
 }
 
+void CThread::TermHandler() { }
+
 void CThread::Sleep(unsigned int milliseconds)
 {
   if(milliseconds > 10 && IsCurrentThread())
     m_StopEvent.WaitMSec(milliseconds);
   else
-    XbmcThreads::ThreadSleep(milliseconds);
+    std::this_thread::sleep_for(std::chrono::milliseconds(milliseconds));
+}
+
+bool CThread::Join(unsigned int milliseconds)
+{
+  CSingleLock l(m_CriticalSection);
+  std::thread* lthread = m_thread;
+  if (lthread != nullptr) {
+    if (IsCurrentThread())
+      return false;
+
+    {
+      CSingleExit exit(m_CriticalSection); // don't hold the thread lock while we're waiting
+      std::future_status stat = m_future.wait_for(std::chrono::milliseconds(milliseconds));
+      if (stat != std::future_status::ready)
+        return false;
+    }
+
+    // it's possible it's already joined since we released the lock above.
+    if (lthread->joinable())
+      m_thread->join();
+    return true;
+  } else
```

Comment:
this is none of your Java code :)
please check entire pr, there is more incorrect coding style